### DISCUSSION TITLE
autohostlist + zapret_check

### DISCUSSION THREAD

**Discussion by numbereleven-a** [bol-van/zapret#364] [Ideas]
А нельзя ли сделать режим работы autohostlist с учётом 4ой проверки на доступность с использованием конфига запрета? 
Пример
1. Сайт 123.com недоступен
2. Проходит fail counter 1/3, 2/3, 3/3
3. Сайт 123.com добавляется на проверку доступности с применением используемого конфига
4. Если помогает то уже добавлять сайт zapret-hosts-auto.txt
5. Если не помогает то добавлять в файл с пометкой времени например на 10мин, час или сутки (чтобы давать шанс на дурение на периодической основе)

По 5му пункту не уверен, может быть какая то другая более лучшая логика есть.

**Comment by bol-van** [comment]
это чтобы уменьшить  false positives ?
в принципе возможно
пункт 4
но слишком усложнять логику не хотелось бы

**Reply by numbereleven-a** [reply]
ну да
согласен, с добавлением в zapret-hosts-auto.txt всё просто, добавляется 1 шаг проверкой дурения, а вот как 5ый пункт красиво реализовать чтобы не сильно усложнять надо подумать

**Reply by bol-van** [reply]
не нужен этот огород
важно только отличить случай, когда без обхода стабильно нет, с ним с первого раза да
если нет, значит стратегия не работает
и тут уже юзеру надо думать как ее править

**Reply by numbereleven-a** [reply]
тебе видней
просто было бы хорошо иметь меньше добавлений не рабочих сайтов в zapret-hosts-auto.txt

**Reply by bol-van** [reply]
запиши в эксклуд все критически важное
остальное убирай из авто только по необходимости
хорошо подобранные стратегии мало ломают
а если плохо подобрано замучаешься вести листы

**Reply by numbereleven-a** [reply]
так и делаю, с этим нет проблем
подумал что было бы хорошо уменьшить шанс на попадание в ауто лист сайтов которые в целом не работают, а не потому что они заблокированы, но дурение не помогает
ну ок

**Reply by bol-van** [reply]
вариант с однократным чекингом с включенным обходом
сделаю наверно

**Reply by numbereleven-a** [reply]
> вариант с однократным чекингом с включенным обходом сделаю наверно

Вопрос про zapret-hosts-auto-debug.log
Как то можно в лог добавить имя клиента с которого запрос идёт?

**Reply by bol-van** [reply]
ip:source_port можно
имя нет

**Reply by numbereleven-a** [reply]
> ip:source_port можно имя нет

Попытался разобраться, понял что это делается не одной строчкой в каком то файле конфигурации.
А вы не хотите добавить данный функционал в будущую версии, может быть отдельной опцией которую было бы удобно включить в файле конфига, если считает что по умолчанию она была бы лишняя?

**Reply by bol-van** [reply]
это полезная инфа в хостлист логе
когда много клиентов знать кто что дергает
сейчас такой функции нет

**Reply by numbereleven-a** [reply]
> это полезная инфа в хостлист логе когда много клиентов знать кто что дергает сейчас такой функции нет

вижу что переделка идёт большая сейчас, просто напоминаю, может под руку попадёт реализация добавления ip клиента в лог zapret-hosts-auto-debug.log
и режим добавления сайта в zapret-hosts-auto.txt если он стал доступен после применения стратегии дурения

**Reply by numbereleven-a** [reply]
> это полезная инфа в хостлист логе когда много клиентов знать кто что дергает сейчас такой функции нет

пока нет времени добавить такой функционал?

**Reply by bol-van** [reply]
с IP клиента это не сложно, а вот анализ результата - там все не так просто
довольно сложное программирование

сейчас в приоритете другое
множественные custom листы (custom.d) и user mode ipset

**Reply by numbereleven-a** [reply]
> с IP клиента это не сложно, а вот анализ результата - там все не так просто довольно сложное программирование
> 
> сейчас в приоритете другое множественные custom листы (custom.d) и user mode ipset

ну было бы хорошо внедрить пока показ ip, если это быстро
спасибо
